include/envoy/network/filter.h

```diff
@@ -331,6 +331,13 @@ class FilterChain {
    * const std::vector<FilterFactoryCb>& a list of filters to be used by the new connection.
    */
   virtual const std::vector<FilterFactoryCb>& networkFilterFactories() const PURE;
+
+  /**
+   * @return int64_t The tag of the FilterChainFactoryContext which the FilterChain is built from.
+   * The default implementation is provided for those FilterChain not associated with
+   * FilterChainFactoryContext.
+   */
+  virtual int64_t getTag() const { return 0; }
```

Comment:
Nit: make this `PURE` and fill in the implementations.

Reply:
done